File: C++/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    cin.tie(0);
    ios::sync_with_stdio(0);
    
    unordered_multiset<string> set(participant.begin(), participant.end());
    for(string& c : completion) {
        auto it = set.find(c);
        if(it != set.end()) {
            set.erase(it);
        }
    }
    
    return *set.begin();
}

File: Java/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        for(String s : participant) {
            map.compute(s, (k, v) -> v == null ? 1 : v + 1);
        }
        for(String s : completion) {
            if(map.get(s) > 1) {
                map.put(s, map.get(s) - 1);
            }
            else {
                map.remove(s);
            }
        }
        for(String s : map.keySet()) {
            return s;
        }
        
        return "";
    }
}